src/Http/Middleware/IdpMiddleware.php

```diff
@@ -3,83 +3,22 @@
 namespace Zanichelli\IdpExtension\Http\Middleware;
 
 use Closure;
-use GuzzleHttp\Client;
 use Illuminate\Http\Request;
-use Illuminate\Support\Facades\Log;
 use Illuminate\Support\Facades\Auth;
-use Zanichelli\IdpExtension\Models\Grant;
-use Zanichelli\IdpExtension\Models\Mongodb\Grant as MongoGrant;
-use Zanichelli\IdpExtension\Models\ZUser;
+use Illuminate\Support\Facades\DB;
 use Zanichelli\IdpExtension\Models\ZTrait\ZUserBuilder;
 
-class IdpMiddleware
+class IdpMiddleware extends IdpUserMiddleware
 {
-
     use ZUserBuilder;
 
     public function handle(Request $request, Closure $next)
     {
+        parent::handle($request, $next);
 
-        // Check if the request has the token field
-        if ($request->input('token')) {
-            $token = $request->input('token');
-
-            $client = new Client(['verify' => false]);
-
-            try {
-                $response = $client->get(env('IDP_TOKEN_URL'), [
-                    'headers' => [
-                        'Authorization' => 'Bearer ' . $token
-                    ]
-                ]);
-            } catch (\Exception $e) {
-                Log::error($e->getMessage());
-                if ($request->ajax() || $request->wantsJson()) {
-                    return response()->json([], 401);
-                }
-
-                return redirect(env('IDP_URL') . '?redirect=' . $request->url());
-            }
-
-            if ($response->getStatusCode() == 200) {
-
-                $userJson = \GuzzleHttp\Utils::jsonDecode($response->getBody());
+        $user = Auth::user();
 
-                $roles = $this->createRoleArray($userJson->roles);
-
-                $attributes = $this->createAttributeArray($userJson->attributes);
-
-                $permissions = $this->retrievePermissions($userJson->id, $roles);
-
-                $user = ZUser::create(
-                    $userJson->id,
-                    $userJson->username,
-                    $userJson->email,
-                    $token,
-                    $userJson->is_verified,
-                    $userJson->name,
-                    $userJson->surname,
-                    $userJson->is_employee,
-                    $userJson->created_at,
-                    $roles,
-                    $permissions,
-                    $attributes
-                );
-
-                $this->addExtraParametersToUser($user);
-
-                Auth::setUser($user);
-            }
-        }
-
-        // Check if the user is logged in
-        if (!Auth::user()) {
-            if ($request->ajax() || $request->wantsJson()) {
-                return response()->json([], 401);
-            }
-
-            return redirect(env('IDP_URL') . '?redirect=' . $request->url());
-        }
+        $user->permissions = $this->retrievePermissions($user->id, $user->roles);
```

Comment:
Non sarebbe sufficiente richiamare `$this->retrievePermissions` all'interno di `addExtraParametersToUser` ? Visto che nella classe base è vuoto, potrebbe essere ridefinito qui e integrato con il recupero dei permessi.